### PR TITLE
Verbose message when bios_grub partition is missing (bnc#917763)

### PR DESCRIPTION
**Pull request by ancorgs**

### CODE REVIEW

File: src/include/partitioning/custom_part_check_generated.rb [hunk 1]
```diff
@@ -336,9 +336,12 @@ def check_created_partition_table(targetMap, installation)
       if (!boot_found && installation && 
           !Partitions.EfiBoot && rootdlabel=="gpt") || show_all_popups
         message = _(
-          "Warning: There is no partition of type bios_grub present.\n" +
-          "To boot from a GPT disk using grub2 such a partition is needed.\n" +
-          "\n" +
+          "Warning: There is no partition of type bios_grub present.\n" \
+          "When installing Grub2 to the MBR of a GPT formatted disc it is recommended\n" \
+          "(if Btrfs is used, then it is required) to reserve a small (around 1MB)\n" \
+          "unformatted partition of this type.\n" \
```
Comment (mchf) [inline src/include/partitioning/custom_part_check_generated.rb]: This sentence is killing me - spread over three lines and many braces in it ;-)

Btw this seems to be pretty long for a popup. It looks more like a help text.

Comment (ancorgs) [inline src/include/partitioning/custom_part_check_generated.rb]: Well,the sentence is taken from the bug report. And the report is precisely about having a more informative pop-up. I'll try to make the sentence shorter and more readable without losing the information the reporter wanted to have there.

Comment (ancorgs) [inline src/include/partitioning/custom_part_check_generated.rb]: @mchf What about?
```ruby
"Warning: There is no partition of type bios_grub present.\n" \
"Such a partition is recommended (required with Btrfs) when Grub2 is\n" \
"installed in the MBR of a GPT disk. It must be unformatted and\n" \
"approximately 1MB large." \
"\n" \
"Really use this setup?\n"

File: package/yast2-storage.changes [hunk 2]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Thu Jun 18 09:47:16 UTC 2015 - ancor@suse.com
+
+- Improved the message displayed when a bios_grup partition is
```
Comment (kobliha) [inline package/yast2-storage.changes]: Or maybe `bios_grub`?

File: src/include/partitioning/custom_part_check_generated.rb (hunk 1 above)
Comment (mchf) [inline src/include/partitioning/custom_part_check_generated.rb]: yep, much better. LGTM

File: package/yast2-storage.changes (hunk 2 above)
Comment (ancorgs) [inline package/yast2-storage.changes]: Yep. Good catch

File: src/include/partitioning/custom_part_check_generated.rb [hunk 3]
```diff
@@ -336,9 +336,12 @@ def check_created_partition_table(targetMap, installation)
       if (!boot_found && installation && 
           !Partitions.EfiBoot && rootdlabel=="gpt") || show_all_popups
         message = _(
-          "Warning: There is no partition of type bios_grub present.\n" +
-          "To boot from a GPT disk using grub2 such a partition is needed.\n" +
-          "\n" +
+          "Warning: There is no partition of type bios_grub present.\n" \
+          "When installing Grub2 to the MBR of a GPT formatted disc it is recommended\n" \
```
Comment (kobliha) [inline src/include/partitioning/custom_part_check_generated.rb]: when ... formatted disc>>,<< it is recommended

File: src/include/partitioning/custom_part_check_generated.rb [hunk 4]
```diff
@@ -336,9 +336,12 @@ def check_created_partition_table(targetMap, installation)
...
+          "(if Btrfs is used, then it is required) to reserve a small (around 1MB)\n" \
```
Comment (kobliha) [inline src/include/partitioning/custom_part_check_generated.rb]: (required in case if btrfs partition)

**Comment by kobliha** [comment]
Nice to have such a message but I'm quite lost as well :)
Can't we detect whether, e.g., btrfs is used and then just show different message?
Can we check whether GPT is used, ... can we...?

File: src/include/partitioning/custom_part_check_generated.rb (hunk 3 above)
Comment (aschnell) [inline src/include/partitioning/custom_part_check_generated.rb]: We use "disk", not "disc".

Comment (ancorgs) [inline src/include/partitioning/custom_part_check_generated.rb]: @aschnell, @kobliha You are reviewing the original text that was already agreed to be... let's say... suboptimal. Please review the one proposed in https://github.com/yast/yast-storage/pull/137#discussion_r32722471

**Comment by ancorgs** [comment]
@kobliha, we already know that GPT is used. Nevertheless I find the "of a GPT disk" part informative.

It's more or less the same with btrfs, we could check for the filesystem and write (recommended|required) depending on the case, but I (as a user) would appreciate that the dialog explains me the difference, no matter what I selected.

**Comment by mchf** [comment]
LGTM

**Pull request merged**